etc/docs/api.proto

```diff
@@ -0,0 +1,63 @@
+service ExecutorService{
+  rpc ExecuteCommand (ExecuteRequest) returns (ExecuteReply) {}
+  rpc StopExecution (ExecutionIdRequest) returns (StopCommandReply) {}
+  rpc ListExecutions (EmptyRequest) returns (ExecutionListReply) {}
+  rpc GetExecution (ExecutionIdRequest) returns (ExecutionInfoReply) {}
+  rpc GetOutStream(ExecutionIdRequest) returns (stream OutEntry) {}
+  rpc GetOut(ExecutionIdRequest) returns (OutEntriesReply) {}
+}
+
+message ExecuteRequest {
+  required string command = 1;
+  repeated string args = 2;
+}
+
+message ExecutionIdRequest {
+  required ExecutionId executionId = 1;
+}
+
+message EmptyRequest {}
+
+message ExecutionId {
+  required string value = 1;
+}
+
+message ExecuteReply {
+  required ExecutionId executionId = 1;
+}
+
+message ExecutionListReply {
+  repeated ExecutionInfoReply executions = 1;
+}
+
+message ExecutionInfoReply {
+  required ExecutionId executionId = 1;
+  enum ExecutionStatus {
+    UNKNOWN = 0;
+    FINISHED = 1;
+    RUNNING = 2;
+    ABORTED = 3;
+    NOT_STARTED = 4;
+  }
+  required ExecutionStatus status = 2;
+  optional int32 exitCode = 3;
+}
+
+message StopCommandReply {
+  required ExecutionId executionId = 1;
+  enum StopStatus {
+    UNKNOWN = 0;
+    STOPPED = 1;
+    KILLED = 2;
+  }
```

Comment:
Could you describe shortly how StopExecution will be implemented ?

Reply:
Initially, it will send a SIGTERM to the process, if that works it will return the STOPPED status

If the process is not terminated after some time (probably configurable, for simplicity let's say 3 seconds now) then a SIGKILL will be sent and KILLED status will be returned